### DISCUSSION TITLE
Invoking SignalR client method using JsonDocument instead of object as argument

### DISCUSSION THREAD

**Discussion by oliverw** [dotnet/aspnetcore#28662] [General]
I have a controller that allows other parts of the application which do not run inside a ASPNET core process to send messages to SignalR Clients: 

```
[Route("api/[controller]")]
[ApiController]
public class NotificationController : ControllerBase
{
    public NotificationController(
        IHubContext<NotificationHub, INotificationHub> notificationHub)
    {
        this.notificationHub = notificationHub;
    }

    private readonly IHubContext<NotificationHub, INotificationHub> notificationHub;

    [HttpPost("relay-user-notification")]
    public async Task RelayUserNotification(string userId)
    {
        var group = notificationHub.Clients.Group(NotificationHub.MakeUserGroup(userId));
        using var notification = await JsonDocument.ParseAsync(Request.Body);

        await group.OnNotification(notification);
    }
}
```

```
public interface INotificationHub
{
    Task OnNotification(object notification);
}

```
Since the goal is to support any type of Notification, the controller action simply reads the JSON request data into a System.Text.JsonDocument and fowards it to. This works, but since the implementation is some kind of workaround for #22465 I'd rather ask if this may be some kind of glitch that is not supposed to work.

**Comment by BrennanConroy** [comment]
> I'd rather ask if this may be some kind of glitch that is not supposed to work.

I don't know if I'd call it a glitch, or claim that it shouldn't work. If the serializer can work with the object and you can use it correctly from the client side then that's fine. But it is definitely preferable to use actual types instead of `object`.
